From 1953f220671d4a72c69fe2ec7f8b679f6531deae Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Thu, 29 Sep 2022 19:54:26 +0200
Subject: [PATCH] SSDK: install the kmod properly

Call modules_install with the passed path after compilation so that kmod
gets properly installed by the kernel and thus can be autoloaded and
linked against.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 77936517..5d7699ef 100755
--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,7 @@ SUB_LIB=$(subst src/, , $(SUB_DIR))
 all: $(BIN_DIR) kslib
 	mkdir -p ./temp/;cd ./temp;cp ../build/bin/ssdk_ks_km.a ./;ar -x ssdk_ks_km.a; cp ../ko_Makefile ./Makefile;
 	make -C $(SYS_PATH) M=$(PRJ_PATH)/temp/ CROSS_COMPILE=$(TOOLPREFIX) modules
+	make -C $(SYS_PATH) M=$(PRJ_PATH)/temp/ CROSS_COMPILE=$(TOOLPREFIX) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install
 	cp $(PRJ_PATH)/temp/Module.symvers $(PRJ_PATH)/Module.symvers;
 	cp temp/*.ko build/bin;
 	rm -Rf ./temp/*.o ./temp/*.ko ./temp/*.a
-- 
2.37.3

